internal/export/distro/registrations.go

```diff
@@ -352,3 +355,13 @@ func Loop() {
 		}
 	}
 }
+
+func stop(registrations map[string]*registrationInfo) {
+	for k, reg := range registrations {
+		if !reg.deleteFlag {
```

Comment:
I'm not super familiar with this part of the codebase, but should if the `deleteFlag` is false, then we write something to the channel, then delete it? Or should we just write something to the channel and then continue to the next one? I.e. should this code actually be:

```go
	for k, reg := range registrations {
		if !reg.deleteFlag {
			// Do not write in channel that will not be read
			reg.chRegistration <- nil
                        continue
		}
		delete(registrations, k)
        }
```

Reply:
Well, there's no testing or documentation on this. It looks harmless as the logic is just cleaning out the registrations map before returning. I'm hesitant to make a change here if there isn't a reported issue leading to it. If a message comes through that `messageErrors` channel above, the service dies, so that's the only path that would come through here.